tests/static_map/key_sentinel_test.cu

```diff
@@ -60,25 +64,23 @@ TEMPLATE_TEST_CASE_SIG(
   {
     REQUIRE(cuco::test::all_of(pairs_begin,
                                pairs_begin + num_keys,
-                               [m_view] __device__(cuco::pair<Key, Value> const& pair) mutable {
-                                 return m_view.insert(
-                                   pair, cuco::default_hash_function<Key>{}, custom_equals<Key>{});
+                               [insert_ref] __device__(cuco::pair<Key, Value> const& pair) mutable {
+                                 return insert_ref.insert(pair);
                                }));
   }
 
   SECTION(
     "Tests of CG insert: The custom `key_equal` can never be used to compare against sentinel")
   {
-    map.insert(pairs_begin,
-               pairs_begin + num_keys,
-               cuco::default_hash_function<Key>{},
-               custom_equals<Key>{});
+    map.insert(pairs_begin, pairs_begin + num_keys);
     // All keys inserted via custom `key_equal` should be found
-    REQUIRE(cuco::test::all_of(
-      pairs_begin, pairs_begin + num_keys, [view] __device__(cuco::pair<Key, Value> const& pair) {
-        auto const found = view.find(pair.first);
-        return (found != view.end()) and
-               (found->first.load() == pair.first and found->second.load() == pair.second);
-      }));
+    REQUIRE(cuco::test::all_of(pairs_begin,
+                               pairs_begin + num_keys,
+                               [find_ref] __device__(cuco::pair<Key, Value> const& pair) {
```

Comment:
Do we need to specify the return type here via `cuda::proclaim_return_type`?

Reply:
good point. Let me do this to make #343 easier.